File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>

// Prototype functions
int counters(string text);
void printGrade(int);
int colemanIndex(int, int, int);

// Global variables
int l, w, s, index;

int main(void)
{
    // Gets text from the user
    string text = get_string("Text: ");
    // Analyzes the text
    counters(text);
    colemanIndex(l, w, s);
    // Print results
    printGrade(index);
}
// Iterates through all the text and counts each letter, word and sentence.
int counters(string text)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]) > 0)
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return l = letters, w = words, s = sentences;
}
// Uses the Coleman index formula to grade the text
int colemanIndex(int a, int b, int c)
{
    return index = round(0.0588 * (a / ((float)b / 100)) - 0.296 * (c / ((float)b / 100)) - 15.8);
}
// Printing different outputs based on grade
void printGrade(int grade)
{
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Starting variables, words starts as 1 cus we count a word per space
letters = sentences = 0
words = 1

# Counting
for item in text:
    if item.isalpha():
        letters += 1
    if item.isspace():
        words += 1
    if item in ["?", ".", "!"]:
        sentences += 1

# Applying formula
L = (letters * 100) / words
S = (sentences * 100) / words

result = round(0.0588 * L - 0.296 * S - 15.8)

# Printing results
if result < 1:
    print("Before Grade 1")
    
elif result >= 16:
    print("Grade 16+")
    
else:
    print(f"Grade {result}")